localstack/services/stepfunctions/asl/component/state/state_wait/wait_function/seconds_path.py

```diff
@@ -18,6 +17,10 @@ def __init__(self, path: str):
         self.path: Final[str] = path
 
     def _get_wait_seconds(self, env: Environment) -> int:
-        input_expr = parse(self.path)
-        seconds = input_expr.find(env.inp)
+        inp = env.stack[-1]
+        seconds = JSONPathUtils.extract_json(self.path, inp)
+        if not isinstance(seconds, int) and seconds <= 0:
+            raise ValueError(
+                f"Expected non-negative integer for SecondsPath, got '{type(seconds).__name__}' '{seconds}' instead."
```

Comment:
Are these error messages specific to LocalStack or did you validate them against AWS?

Reply:
Thank you, I decided to add some negative tests and raise the appropriate validation exceptions. These also match the event history outputs from AWS